src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -422,6 +422,11 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('compressSingleChildFolders', "Controls whether the explorer should render folders in a compact form. In such a form, single child folders will be compressed in a combined tree element. Useful for Java package structures, for example."),
 			'default': true
 		},
+		'explorer.collapseWorkspaceFoldersFirst': {
```

Comment:
I am not a fan of adding settings for features which I think just work nicely out of the box.
I suggest to remove this setting, and only consider adding it if users complain.

IMHO the change you did is intuitive behavior and will not cause a big user backlash.

Reply:
Sounds good to me, removed